src/java.base/share/classes/java/lang/Shutdown.java

```diff
@@ -157,16 +157,35 @@ static void halt(int status) {
      * which should pass a nonzero status code.
      */
     static void exit(int status) {
+        System.Logger log = getRuntimeExitLogger(); // Locate the logger without holding the lock;
         synchronized (Shutdown.class) {
             /* Synchronize on the class object, causing any other thread
              * that attempts to initiate shutdown to stall indefinitely
              */
+            if (log != null) {
+                Throwable throwable = new Throwable("Runtime.exit(" + status + ")");
+                log.log(System.Logger.Level.DEBUG, "Runtime.exit() called with status: " + status,
+                        throwable);
```

Comment:
I'd put a try/catch around the actual logging of the message, to prevent an error in the logger handler from preventing the system from being shut down.

Reply:
If there is an error in the logging, it should not be hidden so it can be fixed by the developer or logging configuration.
A mis-configured or malfunctioning logging system might hide more than just this particular case.